queries/editorconfig/highlights.scm

```diff
@@ -0,0 +1,56 @@
+(comment) @comment @spell
+
+; consistency with toml and ini
+(section
+  (section_name) @type)
+
+(character_choice
+  (character) @constant)
+
+(character_range
+  start: (character) @constant
+  end: (character) @constant)
+
+[
+  "["
+  "]"
+  "{"
+  "}"
+] @punctuation.bracket
+
+[
+  ","
+  ".."
+  (path_separator)
+] @punctuation.delimiter
+
+[
+  "-"
+  "="
+  (negation)
+] @operator
+
+[
+  (wildcard_characters)
+  (wildcard_any_characters)
+  (wildcard_single_character)
+] @character.special
```

Comment:
These also apply in section headings -- is that intended? In fact, these together with `character` completely override the `section_name`.

Reply:
Pretty much intended to apply them in headings, actually that is the *only* place where they appear.

The `character` node is not captured in headings for reasons that I mentioned in another comment (https://github.com/nvim-treesitter/nvim-treesitter/pull/6855#discussion_r1665553487). In retrospective, maybe I shouldn't have exposed `character` that much.

Reply:
> The character node is not captured in headings for reasons that I mentioned in another comment (https://github.com/nvim-treesitter/nvim-treesitter/pull/6855#discussion_r1665553487).

But it _is_: see the following example.
```editorconfig
[*.py]
indent_size = 4
```
Tree:
```
  (section ; [12, 0] - [15, 0]
    (section_name ; [12, 1] - [12, 5]
      (wildcard_characters) ; [12, 1] - [12, 2]
      (character) ; [12, 2] - [12, 3]
      (character) ; [12, 3] - [12, 4]
      (character)) ; [12, 4] - [12, 5]
    (pair ; [13, 0] - [14, 0]
      key: (identifier) ; [13, 0] - [13, 11]
      value: (integer))) ; [13, 14] - [13, 15]
```

Again, trying to understand the intention here to see if the queries can (should) be improved.

Reply:
Yeah, if they're not meant to be captured, it's better to keep them private (underscored or inlined).

Reply:
(It just looks weird to me, but if that's what people expect, I don't mind.)